### PR TITLE
EZP-26804: Search with section id list filter

### PR DESCRIPTION
**Pull request by blankse**
JIRA: https://jira.ez.no/browse/EZP-26804

### CODE REVIEW

File: classes/ezfezpsolrquerybuilder.php [hunk 1]
```diff
@@ -1530,6 +1531,53 @@ protected function getContentClassFilterQuery( $contentClassIdent )
     }
 
     /**
+     * Generate section query filter.
+     *
+     * @param mixed eZSection id or list of ids.
```
Comment (bdunogier) [inline classes/ezfezpsolrquerybuilder.php]: `@param int|int[] $sectionIdent eZSection id or list of ids.`

File: classes/ezfezpsolrquerybuilder.php [hunk 2]
```diff
@@ -1530,6 +1531,53 @@ protected function getContentClassFilterQuery( $contentClassIdent )
...
+     *
+     * @return string Section query filter. Returns null if invalid
+     * $sectionIdent is provided.
```
Comment (bdunogier) [inline classes/ezfezpsolrquerybuilder.php]: You need to align the second line with `string`.

File: classes/ezfezpsolrquerybuilder.php [hunk 3]
```diff
@@ -1530,6 +1531,53 @@ protected function getContentClassFilterQuery( $contentClassIdent )
...
+                }
+                else
+                {
+                    eZDebug::writeError( "Unknown section_id filtering parameter: $sectionID", __METHOD__ );
+                }
+            }
+
+            return implode( ' OR ', $sectionQueryParts );
+        }
+        elseif ( (int)$sectionIdent . '' == $sectionIdent . '' )
```
Comment (bdunogier) [inline classes/ezfezpsolrquerybuilder.php]: I'd handle this case first as it is the most common. Then I'd get rid of the `else`, as we return inside the if.

**Comment by blankse** [comment]
@bdunogier I take your remarks into account :)

File: classes/ezfezpsolrquerybuilder.php [hunk 4]
```diff
@@ -1530,6 +1531,53 @@ protected function getContentClassFilterQuery( $contentClassIdent )
...
+        if ( empty( $sectionIdent ) )
+        {
+            return null;
+        }
+
+        if ( is_numeric( $sectionIdent ) )
+        {
+            return eZSolr::getMetaFieldName( 'section_id' ) . ':' . $sectionIdent;
+        }
+        elseif ( is_array( $sectionIdent ) )
```
Comment (bdunogier) [inline classes/ezfezpsolrquerybuilder.php]: You don't need the `else`, since you return in the `if` above.

**Comment by blankse** [comment]
@bdunogier I removed the else

**Comment by blankse** [comment]
Ping @bdunogier

File: classes/ezfezpsolrquerybuilder.php [hunk 5]
```diff
@@ -1529,6 +1530,54 @@ protected function getContentClassFilterQuery( $contentClassIdent )
...
+            return eZSolr::getMetaFieldName( 'section_id' ) . ':' . $sectionIdent;
+        }
+
+        if ( is_array( $sectionIdent ) )
+        {
+            $sectionQueryParts = array();
+            foreach ( $sectionIdent as $sectionID )
+            {
+                $sectionID = trim( $sectionID );
+                if ( empty( $sectionID ) )
```
Comment (andrerom) [inline classes/ezfezpsolrquerybuilder.php]: Is empty array value something we need to support? it could just go straight to error, or?

Comment (blankse) [reply classes/ezfezpsolrquerybuilder.php]: I can remove it

File: classes/ezfezpsolrquerybuilder.php [hunk 6]
```diff
@@ -1529,6 +1530,54 @@ protected function getContentClassFilterQuery( $contentClassIdent )
...
+     *         $sectionIdent is provided.
+     */
+    protected function getSectionFilterQuery( $sectionIdent )
+    {
+        if ( empty( $sectionIdent ) )
+        {
+            return null;
+        }
+
+        if ( is_numeric( $sectionIdent ) )
```
Comment (gggeek) [inline classes/ezfezpsolrquerybuilder.php]: might be better to use ctype_digit()

**Comment by gggeek** [comment]
Beside remark, +1

**Comment by blankse** [comment]
@andrerom @gggeek I changed it.

**Pull request merged**